app/controllers/notes_controller.rb

```diff
@@ -69,6 +69,10 @@ def show
       if current_user == nil || (current_user.role != 'admin' && current_user.role != 'moderator')
         redirect_to DrupalNode.find(@node.power_tag('redirect')).path
         return
+      elsif (current_user.role == 'admin' || current_user.role == 'moderator')
+        flash.now[:warning] = "Only moderators and admins see this page, as it is redirected to #{DrupalNode.find(@node.power_tag('redirect')).title}.
+        To remove the redirect, delete the tag beginning with 'redirect:'"
+        return
```

Comment:
I think this `return` is probably unnecessary -- i think it's in the line above because we can't both redirect and render, so we "end" the method execution by saying `return` after `redirect_to`.